main.cpp

```diff
@@ -96,6 +104,19 @@ struct hash<address_v4::bytes_type> : hash<uint32_t>
 	}
 };
 
+template <>
+struct hash<ipv6_prefix_type> : hash<uint32_t>
+{
+	size_t operator()(ipv6_prefix_type ip) const
+	{
+		// this is the crc32c (Castagnoli) polynomial
+		boost::crc_optimal<32, 0x1EDC6F41, 0xFFFFFFFF, 0xFFFFFFFF, true, true> crc;
```

Comment:
just fyi. if people spoof their IPv6 addresses, they may launch a hash-collision attack (since this hash function is well know). The typical solution to this is to use "sip hash" and randomize a secret key on startup.